src/server/v2.0/harbor/handler/handler.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/go-openapi/errors"
+	"github.com/goharbor/harbor/src/server/middleware"
+	"github.com/goharbor/harbor/src/server/v2.0/harbor/handler/artifact"
+	"github.com/goharbor/harbor/src/server/v2.0/harbor/restapi"
+	"github.com/justinas/alice"
+)
+
+// New returns http handler for API V2.0
+func New() http.Handler {
+	h, api, err := restapi.HandlerAPI(restapi.Config{
+		InnerMiddleware: innerMiddleware,
+		ArtifactAPI:     artifact.New(),
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	api.ServeError = serverError
+
+	return h
+}
+
+func innerMiddleware(next http.Handler) http.Handler {
+	return alice.New(middleware.RequestID()).Then(next)
+}
+
+type apiError struct {
+	Code    int32  `json:"code"`
+	Message string `json:"message"`
+}
+
+type ociError struct {
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
+// transform openapi error response to OCI error response format
+func serverError(rw http.ResponseWriter, r *http.Request, err error) {
+	w := httptest.NewRecorder()
+	errors.ServeError(w, r, err)
```

Comment:
What is this used for?
Shouldn't we get the HTTP status code by parsing the err directly?
If it is, please use the common method here https://github.com/goharbor/harbor/blob/master/src/server/error/error.go#L36

Reply:
go-swagger will bind a params object to a request and validate the request, when this step failed it will return directly. 

`ServeError` will handle this problem, the default impelmention will return `{"code": 4xx, "message": "xxxxx" }` format json as response.

`serveError` here used to transform the format below to our common error format.

Reply:
Changed to return our `common error` rather than OCI error.